File: app/src/main/java/com/smarthub/launcher/MainActivity.java
package com.smarthub.launcher;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.Toast;

public class MainActivity extends Activity {

    private final String PLUTO_PACKAGE_NAME = "tv.pluto.android";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // Garanta que o nome do XML aqui seja o mesmo que está na pasta res/layout
        setContentView(R.layout.main); 

        Button btnOpenPluto = findViewById(R.id.btnOpenPluto);
        LinearLayout btnAppPluto = findViewById(R.id.btnAppPluto);

        View.OnClickListener plutoListener = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                openPlutoTV();
            }
        };

        if (btnOpenPluto != null) btnOpenPluto.setOnClickListener(plutoListener);
        if (btnAppPluto != null) btnAppPluto.setOnClickListener(plutoListener);
    }

    private void openPlutoTV() {
        Intent launchIntent = getPackageManager().getLaunchIntentForPackage(PLUTO_PACKAGE_NAME);
        if (launchIntent != null) {
            startActivity(launchIntent);
        } else {
            try {
                Intent webIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://pluto.tv/pt/live-tv"));
                startActivity(webIntent);
            } catch (Exception e) {
                Toast.makeText(this, "Pluto TV não instalada", Toast.LENGTH_SHORT).show();
            }
        }
    }

    @Override
    public void onBackPressed() {
        // Trava o botão voltar para o app agir como launcher
    }
}

File: MainActivity.kt
package com.smarthub.launcher

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.widget.Button
import android.widget.LinearLayout
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private val PLUTO_PACKAGE_NAME = "tv.pluto.android"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btnOpenPluto = findViewById<Button>(R.id.btnOpenPluto)
        val btnAppPluto = findViewById<LinearLayout>(R.id.btnAppPluto)
        val btnAppAll = findViewById<LinearLayout>(R.id.btnAppAll)

        // Ação para abrir a Pluto TV no destaque principal
        btnOpenPluto.setOnClickListener {
            openPlutoTV()
        }

        // Ação ao clicar no ícone do app Pluto TV no rodapé
        btnAppPluto.setOnClickListener {
            openPlutoTV()
        }

        // Ação para abrir a lista completa de apps do sistema
        btnAppAll.setOnClickListener {
            openAllApps()
        }
    }

    private fun openPlutoTV() {
        val launchIntent = packageManager.getLaunchIntentForPackage(PLUTO_PACKAGE_NAME)
        if (launchIntent != null) {
            // App instalado: Abre o aplicativo da Pluto TV
            startActivity(launchIntent)
        } else {
            // App não instalado: Tenta abrir via navegação Web ou Play Store
            try {
                val webIntent = Intent(Intent.ACTION_VIEW, Uri.parse("https://pluto.tv/pt/live-tv"))
                startActivity(webIntent)
            } catch (e: Exception) {
                Toast.makeText(this, "Pluto TV não instalada", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun openAllApps() {
        // Abre a lista padrão de apps do Android
        val intent = Intent(Intent.ACTION_SETTING_PREFERENCES)
        startActivity(intent)
    }

    // Bloqueia a tecla "Voltar" para o launcher não fechar sozinho na TV Box
    override fun onBackPressed() {
        // Não faz nada (mantém o usuário na Launcher)
    }
}
